packages/loader/container-loader/src/protocol/protocol.ts

```diff
@@ -125,7 +125,9 @@ export class ProtocolOpHandler implements IProtocolHandler {
 			}
 
 			case MessageType.Propose: {
-				// This should become unconditional once DeltaManager.processInboundMessage() stops parsing content (ADO #12052)
+				// DeltaManager is no longer parsing the contents, so we expect this to be a string here
```

Comment:
What about the `ClientJoin` case above? It's trying to JSON.parse the contents.data. That will fail if the contents is not parsed right?

Reply:
It's parsing `message.data`, that's independent from `message.contents`.  `data` is always provided by the service, in contrast to `contents` which comes from the client.